app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/Indexer/ModifySelectInProductPriceIndexFilter.php

```diff
@@ -95,6 +105,18 @@ public function aroundModifyPrice(
                     []
                 );
                 $select->where('inventory_stock.is_salable = 0 OR inventory_stock.is_salable IS NULL');
+            } else {
+                $legacyStockTableName = $this->resourceConnection->getTableName('cataloginventory_stock_status');
+                $select->joinLeft(
+                    ['stock_status' => $legacyStockTableName],
+                    sprintf(
+                        'stock_status.product_id = price_index.%s and stock_status.website_id = %d',
```

Comment:
Website ID is also added later. Should we just join on sku in inventory and product?